pdns/recursordist/nod.cc

```diff
@@ -59,27 +59,27 @@ bool PersistentSBF::init(bool ignore_pid)
 {
   auto log = g_slog->withName("nod");
   std::lock_guard<std::mutex> lock(d_cachedir_mutex);
-  if (d_cachedir.length()) {
-    filesystem::path p(d_cachedir);
+  if (d_cachedir.length() != 0) {
+    filesystem::path path(d_cachedir);
     try {
-      if (filesystem::exists(p) && filesystem::is_directory(p)) {
-        remove_tmp_files(p, lock);
+      if (filesystem::exists(path) && filesystem::is_directory(path)) {
+        remove_tmp_files(path, lock);
         filesystem::path newest_file;
-        std::time_t newest_time = time(nullptr);
+        std::time_t newest_time = 0;
         Regex file_regex(d_prefix + ".*\\." + bf_suffix + "$");
-        for (filesystem::directory_iterator i(p); i != filesystem::directory_iterator(); ++i) {
-          if (filesystem::is_regular_file(i->path()) && file_regex.match(i->path().filename().string())) {
-            if (ignore_pid || (i->path().filename().string().find(std::to_string(getpid())) == std::string::npos)) {
+        for (const auto& file : filesystem::directory_iterator(path)) {
+          if (filesystem::is_regular_file(file.path()) && file_regex.match(file.path().filename().string())) {
+            if (ignore_pid || (file.path().filename().string().find(std::to_string(getpid())) == std::string::npos)) {
               // look for the newest file matching the regex
-              if ((last_write_time(i->path()) < newest_time) || newest_file.empty()) {
-                newest_time = last_write_time(i->path());
-                newest_file = i->path();
+              if (last_write_time(file.path()) > newest_time) {
```

Comment:
This new logic is definitely correct :)

Reply:
BTW the comment for this function could probably do with updating, it currently states:
```
// This looks for an old (per-thread) snapshot. The first one it finds,
// it restores from that.
```
I think it should say something like:
```
// This looks for the newest (per-thread) snapshot it can find,
// and restores from that.
```